mage_ai/cache/block.py

```diff
@@ -43,17 +47,22 @@ def build_key(self, block: Union[Dict], repo_path: str = None) -> str:
         """
         block_type = ''
         block_uuid = ''
+        block_language = ''
         configuration = None
         file_path = None
         repo_path = repo_path or get_repo_path(root_project=False)
 
         if isinstance(block, dict):
             block_type = block.get('type')
             block_uuid = block.get('uuid')
+            block_language = block.get('language')
             configuration = block.get('configuration') or {}
+            block_type_plural = inflection.pluralize(block_type) \
+                if block_type != BlockType.CUSTOM \
```

Comment:
Custom blocks specifically should not be pluralized because the folder containing the custom blocks is `custom`, not `customs`.